rust-runtime/aws-smithy-runtime-api/src/client/runtime_components.rs

```diff
@@ -617,13 +639,13 @@ impl RuntimeComponentsBuilder {
         }
 
         Self::new("aws_smithy_runtime_api::client::runtime_components::RuntimeComponentBuilder::for_tests")
-            .with_auth_scheme(SharedAuthScheme::new(FakeAuthScheme))
-            .with_auth_scheme_option_resolver(Some(SharedAuthSchemeOptionResolver::new(FakeAuthSchemeOptionResolver)))
-            .with_endpoint_resolver(Some(SharedEndpointResolver::new(FakeEndpointResolver)))
-            .with_http_connector(Some(SharedHttpConnector::new(FakeConnector)))
-            .with_identity_resolver(AuthSchemeId::new("fake"), SharedIdentityResolver::new(FakeIdentityResolver))
+            .with_auth_scheme(FakeAuthScheme)
+            .with_auth_scheme_option_resolver(Some(FakeAuthSchemeOptionResolver))
+            .with_endpoint_resolver(Some(FakeEndpointResolver))
+            .with_http_connector(Some(FakeConnector))
```

Comment:
should these methods accept `T` instead of `Option<T>`? I might imagine a `set_` that accepted an option 💭

Reply:
It already has the the `set_` variants. I can't remember why I did it this way, but it looks like nothing is calling them with `None`. So I think this is a sane change to make.

Reply:
Thinking this should be done in a follow-up PR.